### PR TITLE
Integration testing contact details

### PR DESCRIPTION
**Pull request by mattonfoot**
### Context

Adds form submission tests for Contact Details page

### Changes proposed in this pull request

- update page object
- rewrite test scenarios
- add standard page tests

### CODE REVIEW

File: integration_tests/pages/components/contact-information/contactDetailsForm.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  // ACTIONS
+
+  get saveAndContinueButton(): PageElement {
+    return this.form.get('button[type=submit][value="continue"]')
+  }
+
+  get saveAndReturnButton(): PageElement {
+    return this.form.get('button[type=submit][value="back"]')
+  }
```
Comment (ben-cart3r) [inline integration_tests/pages/components/contact-information/contactDetailsForm.ts:40]: I think every form has these (or at least most) so maybe we abstract these to a base class?

File: integration_tests/pages/components/contact-information/contactDetailsForm.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    return cy.get('form') // [action*="contact-information/contact-details"]
+  }
+
+  checkHasForm(): void {
+    this.form.should('exist')
+  }
+
+  hasAction(action: string | RegExp): PageElement {
+    return this.form.should('have.attr', 'action', action)
+  }
```
Comment (ben-cart3r) [inline integration_tests/pages/components/contact-information/contactDetailsForm.ts:19]: I think every form has these (or at least most) so maybe we abstract these to a base class?

**Review by ben-cart3r — DISMISSED** [review]
👍

**Review by bjpirt — DISMISSED** [review]
Approved, although should we also be testing the data we're sending to the API?

**Comment by ben-cart3r** [comment]
> Approved, although should we also be testing the data we're sending to the API?

I did wonder this too, it feels like testing the implementation. Possibly the tests should be, given I've submitted this data, if I navigate back to the page, is the same data visible but it does rely heaving on correct mocking

**Review by bjpirt — APPROVED** [review]

**Pull request merged**